syntax = "proto3";
package MeetChat;
// 导入一些公共的 protobuf 类型
import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

// 定义一个枚举类型，表示不同的消息类型
enum MessageType {
  TEXT = 0; // 文本消息
  EMOJI = 1; // 表情消息
  IMAGE = 2; // 图片消息
  FILE = 3; // 文件消息
  VIDEO = 4; // 视频帧消息
}

// 定义一个通用的消息类型，可以包含不同类型的数据
message Message {
  // 消息的发送者
  string sender_id = 1;
  // 消息的接收者
  string receiver_id = 2; // -1 表示所有人接受
  // 消息的时间戳
  google.protobuf.Timestamp timestamp = 3;
  // 消息的类型，使用枚举类型来表示
  MessageType type = 4;
  // 消息的数据，可以是任何类型，使用 Any 类型来封装
  google.protobuf.Any data = 5;
}

// 定义一个视频帧类型，包含帧序号、时间戳和图像数据
message VideoFrame {
  // 帧的序号
  int32 frame_number = 1;
  // 帧的时间戳
  google.protobuf.Timestamp timestamp = 2;
  // 帧的图像数据，可以是任何编码格式，例如 JPEG 或 H.264
  bytes image_data = 3;
}

// 定义一个文字消息类型，包含文本内容
message TextMessage {
  // 文本内容
  string text = 1;
}

// 定义一个表情类型，包含表情符号或图片
message Emoji {
  // 表情符号，例如 😊 或 😂
  string symbol = 1;
  // 表情图片，如果有的话，可以是任何编码格式，例如 PNG 或 GIF
  bytes image = 2;
}

// 定义一个图片类型，包含图片名称、大小和数据
message Image {
  // 图片名称，例如 "cat.jpg"
  string name = 1;
  // 图片大小，以字节为单位
  int32 size = 2;
  // 图片数据，可以是任何编码格式，例如 JPEG 或 PNG
  bytes data = 3;
}

// 定义一个文件类型，包含文件名称、大小和数据
message File {
  // 文件名称，例如 "report.pdf"
  string name = 1;
  // 文件大小，以字节为单位
  int32 size = 2;
  // 文件数据，可以是任何格式
  bytes data = 3;
}